File: C_Projects/CC_Euler/src/src/12.cc
#include "CC_Euler.h"

int64 Euler_12()
{
  int tri=0;
  int divisors=0;
  int count=1;
  while ( divisors < 251 )
    {
      tri=tri+count;
      divisors=0;
      int index = 1;
      while ( index*index < tri )
	{
	  if ( tri%index == 0 )
	    {
	      divisors++;
	    }
	  index++;
	}
      count++;
    }
  return (int64)tri;
}

File: Py_Projects/Euler/12.py
def Euler_12(divisors=500):
	'''First triangle number to have over input number of divisors
	'''
	tri=0
	for x in range(1000000):
		tri = tri + x
		count = 2
		root = 2
		while root * root <= tri:
			if tri % root == 0:
				count += 2
			if root * root == tri:
				count -= 1
			root += 1
		if count > divisors:
			return tri

	raise ValueError('range too small')


if __name__ == '__main__':
	print(Euler_12())
